src/config.rs

```diff
@@ -2111,10 +2117,20 @@ pub fn persist_critical_config(config: &TiKvConfig) -> Result<(), String> {
         ));
     }
 
-    // Persist current critical configurations to file.
-    if let Err(e) = config.write_to_file(&last_cfg_path) {
+    // Persist current configurations to temporary file.
+    if let Err(e) = config.write_to_file(&tmp_cfg_path) {
```

Comment:
How about the tmp_file is corrupted last time?

Reply:
if the persisted config file is corrupted, we will get a panic when startup like used to do.

Reply:
To avoid corrupted config file, we can,

1. Write config to a tmp file.
2. Fsync the tmp file.
3. rename the tmp file to the config file.

Reply:
> To avoid corrupted config file, we can,
> 
> 1. Write config to a tmp file.
> 2. Fsync the tmp file.
> 3. rename the tmp file to the config file.

That was what we did here.